File: Ejercicios/CompleteTree/JS/Objects/LinkedList.js
function LinkedList(){
	this.first = null;
	this.add = addLinkedList;
}

	function addLinkedList(value){
		var compare = new Compare();
		if(!this.first){
			this.first = new Node(value);
		}else{
			if(compare.compare(this.first,value)>0){
				var stack = this.first;
				this.first = new Node(value);
				this.first.next = stack;
				return true;
			}else if(compare.compare(this.first,value)==0){
				var stack = this.first.next;
				this.first = new Node(value);
				this.first.next = stack;
			}else{
				var previous = this.first;
				current = this.first.next;

				while(current){
					if(compare.compare(current, value)<0){
						previous = current;
						current = current.next;
					}else if(compare.compare(current, value)>0){
						var stack = current;
						previous.next = new Node(value);
						previous.next.next = stack;
					}else{
						var stack = current.next;
						previous.next = new Node(value);
						previous.next.next = stack;
					}
				}
				previous.next = new Node(value);
				return true;
			}
		}
	} 

File: proyectoParcial/Recursos/PY/Resources/LinkedList.py
# -*- coding: utf-8 -*-

from Resources.Node import *
from Resources.Compare import *

class LinkedList:
    def __init__(self):
        self.first = None

    def addLinkedList(self,value):
        if (not self.first):
            self.first = Node(value)
        else:
            compare = Compare()
            if(compare.compare(self.first,value)>0):
                stack = self.first
                self.first = Node(value)
                self.first.next = stack
                return True
            else:
                previous = self.first
                current = self.first.next

                while(current):
                    if (compare.compare(current,value)<0):
                        previous = current
                        current = current.next
                        return True
                    elif (compare.compare(current,value)>0):
                        stack = current
                        previous.next = Node(value)
                        previous.next.next = stack
                        return True
                    else:
                        previous.next = Node(value)
                        previous.next.next = current.next
                        return True

                previous.next = Node(value)
                return True